ls1intum/Artemis
# `Exam mode`: Fix an issue when deleting exams with multiple test runs by the same instructor

### Checklist
#### General


#### Server

### Motivation and Context
* Fix #6243
* Also fix some issues in server tests and refactor deleting / resetting exams on the server side into its own service

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 exam with quiz, modeling, text and programming exercise

1. Create multiple test runs for the same instructor
2. Delete the exam
3. This should work without issues



### Review Progress


#### Code Review
#### Manual Tests

### Test Coverage
I added new (more realistic) server tests, however the coverage should not be affected
